pkg/template/template.go

```diff
@@ -0,0 +1,102 @@
+// Package init provides utilities to set up new repositories
+package template
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/elhub/gh-dxp/pkg/config"
+)
+
+func Execute(workingDir string, settings *config.Settings) error {
+	// Get the project template URI
+	uri := settings.ProjectTemplateUri
+
+	// Get the current working directory
+
+	// Create the .github directory if it does not exist
+	ghDir := filepath.Join(workingDir, ".github")
+	if _, errStat := os.Stat(ghDir); os.IsNotExist(errStat) {
+		if err := os.Mkdir(ghDir, 0755); err != nil {
+			return fmt.Errorf("could not create .github directory: %w", err)
+		}
+	}
+
+	// Download files
+	files := []struct {
+		fileName  string
+		path      string
+		overwrite bool
+	}{
+		{
+			fileName:  ".editorconfig-template",
+			path:      filepath.Join(workingDir, ".editorconfig"),
+			overwrite: true,
+		},
+		{
+			fileName:  ".gitattributes-template",
+			path:      filepath.Join(workingDir, ".gitattributes"),
+			overwrite: true,
+		},
+		{
+			fileName:  ".gitignore-template",
+			path:      filepath.Join(workingDir, ".gitignore"),
+			overwrite: true,
+		},
+		{
+			fileName:  "README-template.md",
+			path:      filepath.Join(workingDir, "README.md"),
+			overwrite: false,
+		},
+		{
+			fileName:  ".github/CODEOWNERS-template",
+			path:      filepath.Join(ghDir, "CODEOWNERS"),
+			overwrite: false,
+		},
+		{
+			fileName:  ".github/CONTRIBUTING-template.md",
+			path:      filepath.Join(ghDir, "CONTRIBUTING"),
+			overwrite: true,
+		},
+	}
+
+	// Only write file if overwrite = true or file does not exist
+	for _, file := range files {
+		// Check if the file exists
+		_, err := os.Stat(file.path)
+
+		// If the file does not exist or overwrite is true, write the file
+		if os.IsNotExist(err) || file.overwrite {
```

Comment:
Looks to me like we should be checking the unidentified value (_) and not err here?
If this is correct, I find it pretty unintuitive 🤔

Reply:
os.Stat returns:
fileInfo, nil if the file exists.
??, error if the file does not exist.

We are checking that there is no error thrown (so err is not nil) OR that the file should be overwritten. The OR should probably be switched round (so that we check file.overwrite first), I don't think the reverse check on fileInfo is better in this case (check on err is definite, checking on an undefined variable seems risky).

Reply:
Alright, I guess that makes sense.
Is `os.IsNotExist(err)` equivalent to `err = null` in this case?